src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -1106,6 +1107,10 @@ memory::ptr primitive_inst::allocate_output(engine& _engine, memory_pool& pool,
     // For outputs, cpu prim we want to have lockable alloc type
     // Also if the successor of a node is an cpu, then memory needs to be lockable.
     bool is_cpu = _node.get_selected_impl() ? _node.get_selected_impl()->is_cpu() : false;
+    // WA: Since proposal only have cpu impl, if selected_impl of node is empty
+    //     We need to set it to be selected as lockable memory type manually
+    if (_node.is_type<proposal>() && !_node.get_selected_impl())
+        is_cpu = true;
```

Comment:
why this (impl is cpu but _node.get_selected_impl() is empty) can happen?

Reply:
update to select impl for dynamic or static shape during compile_graph as discussed